frontend/test/metabase/scenarios/question/notebook.cy.spec.js

```diff
@@ -153,6 +154,41 @@ describe("scenarios > question > notebook", () => {
     cy.contains(/^Price is less than 5/i);
   });
 
+  it("should show the real number of rows instead of HARD_ROW_LIMIT when loading", () => {
```

Comment:
Awesome.

As a bit of a nitpick, although when we use Cypress code it makes a difference:
we've been trying to treat e2e test code more and more like implementation code in terms of breaking it down.

Some people prefer making small functions, some people use space in a big function.

Below, maybe each block that you almost titled (with a comment) could be an extracted function.

Reply:
Can you point me to some tests with an implementation of the former (making small functions) I can use as a reference? I can definitely see how a some of these actions can be abstracted away

Reply:
@akiselev, 

this file could be an example: https://github.com/metabase/metabase/blob/master/frontend/test/metabase/scenarios/collections/collection-types.cy.spec.js

It may not be very convincing at first because the test functions are still kind of large.

At the same time, if you take a look at the helper functions at the bottom of the file, and how they're used, it may help you get a sense of things.

I didn't find one file with extracted functions that are not reused, and that are used only once and extracted for clarity of each function, including the test function. It's an idea to consider, although Nemanja (our main test fellow) has not embraced this approach fully (and his take has much more weight on the matter than mine).